File: src/app/src/main/java/com/onscreensync/tvapp/utils/UiHelper.java
package com.onscreensync.tvapp.utils;

import android.graphics.Color;
import android.widget.TextView;

public class UiHelper {

    public static void setTextViewFont(TextView textView, String textFont) {
        if(ObjectExtensions.isNullOrEmpty(textFont)) return;

        int toInt = ObjectExtensions.convertToInt(textFont);
        if(toInt == 0) return;

        textView.setTextSize(toInt);
    }
    public static void setTextViewColor(TextView textView, String textColor) {
        if(ObjectExtensions.isNullOrEmpty(textColor)) return;
        textView.setTextColor(parseColor(textColor));
    }

    public static int parseColor(String textColor) {

        if(ObjectExtensions.isNullOrEmpty(textColor)) return Color.WHITE;

        switch (textColor.toLowerCase())
        {
            case "red":
                return Color.RED;
            case "blue":
                return Color.BLUE;
            case "grey":
            case "gray":
                return Color.GRAY;
            case "black":
                return Color.BLACK;
            case "yellow":
                return Color.YELLOW;
            default:
                try {
                    return Color.parseColor(textColor.toLowerCase());
                } catch(Exception ex) {
                    return Color.WHITE;
                }
        }
    }
}

File: src/app/src/main/java/com/onscreensync/tvapp/utils/UiHelper.kt
package com.onscreensync.tvapp.utils

import android.graphics.Color
import android.widget.TextView

object UiHelper {

    @JvmStatic
    fun setTextViewFont(textView: TextView, textFont: String?) {
        if (textFont.isNullOrEmpty()) return

        val toInt = textFont.convertToInt()
        if (toInt == 0) return

        textView.textSize = toInt.toFloat()
    }

    @JvmStatic
    fun setTextViewColor(textView: TextView, textColor: String?) {
        if (textColor.isNullOrEmpty()) return
        textView.setTextColor(parseColor(textColor))
    }

    @JvmStatic
    fun parseColor(textColor: String?): Int {
        if (textColor.isNullOrEmpty()) return Color.WHITE

        return when (textColor!!.lowercase()) {
            "red" -> Color.RED
            "blue" -> Color.BLUE
            "grey", "gray" -> Color.GRAY
            "black" -> Color.BLACK
            "yellow" -> Color.YELLOW
            else -> {
                try {
                    Color.parseColor(textColor.lowercase())
                } catch (ex: Exception) {
                    Color.WHITE
                }
            }
        }
    }
}